x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichLookupService.java

```diff
@@ -85,6 +90,35 @@ protected QueryList queryList(TransportRequest request, SearchExecutionContext c
         };
     }
 
+    private static void validateTypes(DataType inputDataType, MappedFieldType fieldType) {
+        if (inputDataType == DataType.UNSUPPORTED) {
+            throw new EsqlIllegalArgumentException("ENRICH cannot match on unsupported input data type");
+        }
+        if (fieldType == null) {
+            throw new EsqlIllegalArgumentException("ENRICH cannot match on non-existent field");
+        }
```

Comment:
Can these cases really happen? Shouldn't they be prevented at Verification time?
In case, it would be good to have a test for them.